src/extension/modifiers.ts

```diff
@@ -0,0 +1,77 @@
+declare var imports: any;
+declare var global: any;
+
+import {log} from './logging';
+
+const Mainloop = imports.mainloop;
+const Clutter = imports.gi.Clutter;
+
+export enum MODIFIERS_ENUM {
+    SHIFT,
+    CONTROL = 2,
+    ALT = 3,
+    SUPER = 6
+}
+
+export const MODIFIERS: {[key: number]: MODIFIERS_ENUM} = {
+    0: MODIFIERS_ENUM.SHIFT,
+    2: MODIFIERS_ENUM.CONTROL,
+    3: MODIFIERS_ENUM.ALT,
+    6: MODIFIERS_ENUM.SUPER
+};
+
+export default class ModifiersManager {
+    private connected: {[key: string]: Function[]} = {};
+    private modifiers: MODIFIERS_ENUM[] = [];
+
+    private state: any;
+    private previousState: any;
+
+    constructor() {
+        Mainloop.timeout_add(200, this.update.bind(this));
+    }
+
+    public isHolding(modifier: MODIFIERS_ENUM): boolean {
+        return this.modifiers.includes(modifier);
+    }
+
+    public connect(name: string, callback: Function): number {
+        if (this.connected[name] === undefined) {
+            this.connected[name] = [];
+        }
+
+        return this.connected[name].push(callback);
+    }
+
+    private update(): boolean {
+        const [x, y, m] = global.get_pointer();
+
+        if (typeof m !== 'undefined') {
```

Comment:
Why something like `if(!m) ` is not sufficient?

Reply:
for this check `if (m)` would be correct but yes that would be sufficient.